src/pivot-table/data/extract-headers.ts

```diff
@@ -1,19 +1,23 @@
 import { PSEUDO_DIMENSION_INDEX } from "../../constants";
+import type { ExtendedDimensionInfo } from "../../types/QIX";
+import type { HeaderTitle } from "../../types/types";
 
 const extractHeaders = (
   qDim: EngineAPI.INxDimensionInfo[],
   rowCount: number,
   dimensionInfoIndex: number[]
-): (null | string)[][] => {
-  const matrix: (null | string)[][] = Array(dimensionInfoIndex.length)
+): (null | HeaderTitle)[][] => {
+  const matrix: (null | HeaderTitle)[][] = Array(dimensionInfoIndex.length)
     .fill(null)
     .map(() => Array.from({ length: rowCount }, () => null));
 
   dimensionInfoIndex.forEach((dimIndex, colIdx) => {
     if (dimIndex === PSEUDO_DIMENSION_INDEX) {
-      matrix[colIdx][rowCount - 1] = "";
+      matrix[colIdx][rowCount - 1] = { id: "PSEUDO-DIM", title: "" };
```

Comment:
cell.id is been used in `DimensionTitleCell` and wouldn't this cause duplicate key error in react? in case if we have multiple pseudo dim?